quickwit/quickwit-storage/src/metrics.rs

```diff
@@ -41,14 +43,30 @@ pub struct StorageMetrics {
 
 impl Default for StorageMetrics {
     fn default() -> Self {
+        let get_slice_timeout_outcome_total_vec = new_counter_vec(
+            "get_slice_timeout_outcome",
+            "Outcome of get_slice operations. success_after_1_timeout means the operation \
+             succeeded after a retry caused by a timeout.",
+            "storage",
+            &[],
+            ["outcome"],
+        );
+        let get_slice_timeout_successes = [
+            get_slice_timeout_outcome_total_vec.with_label_values(["success_after_0_timeout"]),
+            get_slice_timeout_outcome_total_vec.with_label_values(["success_after_1_timeout"]),
+            get_slice_timeout_outcome_total_vec.with_label_values(["success_after_2+_timeout"]),
```

Comment:
i think these labels look odd, but don't have concrete improvement proposal to make

Reply:
If it's a metric, it should be 1 label. Maybe `attempts`. 
But I am not sure how valuable there is there. 

If I can see number of retries as a metric, total attempts as a metric, and failed attempts as a metric, I have enough to tell me storage is having a problem. 

These labels are just adding noise and bloat to the metrics that I would have to remove anyway

Reply:
@esatterwhite I don't understand your comment.

> If it's a metric, it should be 1 label

If you mean a single label key, this is the case right now. 
`success_after_0_timeout` in here is a label value.
If you mean a single label value, this is like have no label at all.

I am open to change the metrics again, but I would prefer an accurate 
description, that prometheus users won't be at risk of misinterpretation.

Reply:
maybe a single label of `success_after_timeouts=N`. 
It seems like you're wanting a histogram that observes how many timeouts occurred before there was a success.

It feels like this is really 2 different metrics. But it doesn't need to hold up getting the fix out.

Reply:
Merging without change. We can revisit this later if it is not practical.